packages/wdio-sauce-service/src/utils.ts

```diff
@@ -66,12 +68,14 @@ export function makeCapabilityFactory(tunnelIdentifier, options) {
 
         Object.assign(capability, options)
 
-        const sauceOptions = !isLegacy && !isUnifiedPlatform(capability) ? capability['sauce:options'] : capability
-        sauceOptions.tunnelIdentifier = (
-            capability.tunnelIdentifier ||
-            sauceOptions.tunnelIdentifier ||
-            tunnelIdentifier
-        )
+        const sauceOptions: SauceConnectOptions | undefined = !isLegacy && !isUnifiedPlatform(capability) ? capability['sauce:options'] : capability
+        if (sauceOptions) {
+            sauceOptions.tunnelIdentifier = (
+                capability.tunnelIdentifier ||
+                sauceOptions.tunnelIdentifier ||
+                tunnelIdentifier
+            )
+        }
```

Comment:
Why do we check for `sauceOptions`? Seems that `capability['sauce:options']` can be undefined which is handled a bit above. We should remove the check otherwise there is a chance that `tunnelIdentifier` is not being applied.